node/src/main/scala/org/bitcoins/node/Node.scala

```diff
@@ -44,24 +70,33 @@ trait Node extends NodeApi with ChainQueryApi with P2PLogger {
 
   implicit def executionContext: ExecutionContext = system.dispatcher
 
-  val peers: Vector[Peer]
-
-  private val _peerServices: mutable.Map[Peer, ServiceIdentifier] =
-    mutable.Map.empty
+  def peers: Vector[Peer] = peerData.keys.toVector
 
-  def peerServices: Map[Peer, ServiceIdentifier] = _peerServices.toMap
+  def addPeer(peer: Peer): Unit = {
+    if (!_peerData.contains(peer)) {
+      _peerData.put(peer, PeerData(peer, this))
+    }
```

Comment:
nit: maybe add `debug` or `warn` logs in the case that we already have the peer added? It seems like this might be interesting information